From 94732c8831a8503f90996a786e598c6255a2ce99 Mon Sep 17 00:00:00 2001
From: Han Wang <416810799@qq.com>
Date: Wed, 17 Feb 2021 06:17:04 +0100
Subject: [PATCH] Install seccomp filter even if selinux is permissive

 * If selinux is permissive, Google decide to disable seccomp
   filter because debug tools like strace needs both them to be
   disabled to work.

 * This is fine for developers, but if one runs selinux permissive
   in production, it will be super easy for a macilious app to gain
   root access on Q+, with newly interoduced ZygotePreload API.

 * Technical details:
   https://github.com/dantmnf/BrokenSandbox
   https://github.com/vvb2060/Magica

Change-Id: I1a1208a0a82d36aeb49820972429d106919d6bce
Signed-off-by: Albert I <kras@raphielgang.org>
---
 core/jni/com_android_internal_os_Zygote.cpp | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/core/jni/com_android_internal_os_Zygote.cpp b/core/jni/com_android_internal_os_Zygote.cpp
index 9eede83e21e..be00f6383e5 100644
--- a/core/jni/com_android_internal_os_Zygote.cpp
+++ b/core/jni/com_android_internal_os_Zygote.cpp
@@ -127,8 +127,6 @@ static jclass gZygoteClass;
 static jmethodID gCallPostForkSystemServerHooks;
 static jmethodID gCallPostForkChildHooks;
 
-static bool gIsSecurityEnforced = true;
-
 /**
  * True if the app process is running in its mount namespace.
  */
@@ -632,11 +630,6 @@ static void PreApplicationInit() {
 }
 
 static void SetUpSeccompFilter(uid_t uid, bool is_child_zygote) {
-  if (!gIsSecurityEnforced) {
-    ALOGI("seccomp disabled by setenforce 0");
-    return;
-  }
-
   // Apply system or app filter based on uid.
   if (uid >= AID_APP_START) {
     if (is_child_zygote) {
@@ -2198,11 +2191,6 @@ static void com_android_internal_os_Zygote_nativeAllowFileAcrossFork(
 
 static void com_android_internal_os_Zygote_nativeInstallSeccompUidGidFilter(
         JNIEnv* env, jclass, jint uidGidMin, jint uidGidMax) {
-  if (!gIsSecurityEnforced) {
-    ALOGI("seccomp disabled by setenforce 0");
-    return;
-  }
-
   bool installed = install_setuidgid_seccomp_filter(uidGidMin, uidGidMax);
   if (!installed) {
       RuntimeAbort(env, __LINE__, "Could not install setuid/setgid seccomp filter.");
@@ -2280,10 +2268,6 @@ static void com_android_internal_os_Zygote_nativeInitNativeState(JNIEnv* env, jc
    * Security Initialization
    */
 
-  // security_getenforce is not allowed on app process. Initialize and cache
-  // the value before zygote forks.
-  gIsSecurityEnforced = security_getenforce();
-
   selinux_android_seapp_context_init();
 
   /*
-- 
2.30.1

